azurerm/internal/services/storage/storage_management_policy_resource.go

```diff
@@ -413,3 +449,76 @@ func flattenStorageManagementPolicyRules(armRules *[]storage.ManagementPolicyRul
 
 	return rules
 }
+
+func expandAzureRmStorageBlobIndexMatch(blobIndexMatches []interface{}) *[]storage.TagFilter {
+	if len(blobIndexMatches) == 0 {
+		return nil
+	}
+
+	results := make([]storage.TagFilter, 0)
+	for _, v := range blobIndexMatches {
+		blobIndexMatch := v.(map[string]interface{})
+
+		filter := storage.TagFilter{
+			Name:  utils.String(blobIndexMatch["name"].(string)),
+			Op:    utils.String(blobIndexMatch["operation"].(string)),
+			Value: utils.String(blobIndexMatch["value"].(string)),
+		}
+
+		results = append(results, filter)
+	}
+
+	return &results
+}
+
+func flattenAzureRmStorageBlobIndexMatch(blobIndexMatches *[]storage.TagFilter) *schema.Set {
+	result := &schema.Set{
+		F: resourceStorageManagementPolicyBlobIndexMatchHash,
+	}
+
+	if blobIndexMatches == nil || len(*blobIndexMatches) == 0 {
+		return result
+	}
+
+	for _, blobIndexMatch := range *blobIndexMatches {
+		var name, op, value string
+		if blobIndexMatch.Name != nil {
+			name = *blobIndexMatch.Name
+		}
+		if blobIndexMatch.Op != nil {
+			op = *blobIndexMatch.Op
+		}
+		if blobIndexMatch.Value != nil {
+			value = *blobIndexMatch.Value
+		}
+		result.Add(map[string]interface{}{
+			"name":      name,
+			"operation": op,
+			"value":     value,
+		})
+	}
+	return result
+}
+
+func resourceStorageManagementPolicyBlobIndexMatchHash(v interface{}) int {
+	var buf bytes.Buffer
+
+	if m, ok := v.(storage.TagFilter); ok {
```

Comment:
this won't work since this type isn't returned in all circumstances, so this'll be an empty string at plan time - although we shouldn't need a custom hash function here, so can we remove this?